### PR TITLE
Fix GridTileBar layout

### LINKED ISSUE #6836: GridTileBar doesn't seem to respect Material design's spec
## Steps to Reproduce

- Go to Flutter Gallery
- Go to Grid Demo
- (Keep the default "Two Lines")

Expected : All bars should have the same height
Actual: There's a small difference in height between the bottom bars in the tiles "Mountain country" and "Alpine place".

If you want to make it even more visible:

- Change the demo to be "One line"
Expected : All bars should have the same height
Actual: There's a big difference in height between the bottom bars in the tiles "Desert land" and "Narnia".

Note: See [the specs](https://material.google.com/components/grid-lists.html#grid-lists-specs) for specific measurements.

![screenshot_20161113-152933](https://cloud.githubusercontent.com/assets/2762440/20248635/2c6ced4a-a9b6-11e6-9dcd-c8e194badec2.png)

## Logs
N/A

## Flutter Doctor

```
 ❯ flutter doctor                                                                                                                                                                                                        [15:27:09]
[✓] Flutter (on Linux, channel master)
    • Flutter at /home/joel/code/flutter
    • Framework revision c1f817dfa1 (3 hours ago), 2016-11-13 12:30:48
    • Engine revision 84a060820a
    • Tools Dart version 1.21.0-dev.3.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.0)
    • Android SDK at /opt/android-sdk
    • Platform android-25, build-tools 25.0.0
    • ANDROID_HOME = /opt/android-sdk
    • Java(TM) SE Runtime Environment (build 1.8.0_112-b15)

[✓] IntelliJ IDEA Ultimate Edition (version 2016.3)
    • Dart plugin installed
    • Flutter plugin installed

[✓] Connected devices
    • Nexus 6P • 84B7N15A10004400 • android-arm • Android 7.1.1 (API 25)
```

### PR DESCRIPTION
**Pull request by HansMuller**
Force GridTileBar heights to match the Material spec (https://material.google.com/components/grid-lists.html#grid-lists-specs) and simplify their layout a little.

Correct one more Scaffold body demo layout bug.

Fixes #6836

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart [hunk 1]
```diff
@@ -164,10 +164,15 @@ class GridDemoPhotoItem extends StatelessWidget {
           appBar: new AppBar(
             title: new Text(photo.title)
           ),
-          body: new Hero(
-            tag: photo.tag,
-            child: new GridPhotoViewer(photo: photo),
-          )
+          body: new FractionallySizedBox(
+           alignment: FractionalOffset.topLeft,
+            widthFactor: 1.0,
```
Comment (abarth) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: bad indent

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart [hunk 2]
```diff
@@ -164,10 +164,15 @@ class GridDemoPhotoItem extends StatelessWidget {
           appBar: new AppBar(
             title: new Text(photo.title)
           ),
-          body: new Hero(
-            tag: photo.tag,
-            child: new GridPhotoViewer(photo: photo),
-          )
+          body: new FractionallySizedBox(
```
Comment (abarth) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: Maybe SizedBox.expand?

File: packages/flutter/lib/src/material/grid_tile_bar.dart [hunk 3]
```diff
@@ -64,13 +64,9 @@ class GridTileBar extends StatelessWidget {
...
-    EdgeInsets padding;
-    if (leading != null && trailing != null)
-      padding = const EdgeInsets.symmetric(vertical: 16.0, horizontal: 8.0);
-    else if (leading != null)
-      padding = const EdgeInsets.only(left: 8.0, right: 16.0, top: 16.0, bottom: 16.0);
-    else // trailing != null || (leading == null && trailing == null)
-      padding = const EdgeInsets.only(left: 16.0, right: 8.0, top: 16.0, bottom: 16.0);
+    EdgeInsets padding = (leading != null)
+      ? const EdgeInsets.only(left: 8.0, right: 16.0)
+      : const EdgeInsets.only(left: 16.0, right: 8.0);
```
Comment (abarth) [inline packages/flutter/lib/src/material/grid_tile_bar.dart]: Why are there only two cases now?  I Would have expected the `trailing == null` and `trailing != null` cases to be treated differently.

File: examples/flutter_gallery/lib/demo/grid_list_demo.dart (hunk 2 above)
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/grid_list_demo.dart]: Yes, that would be WAY better.

File: packages/flutter/lib/src/material/grid_tile_bar.dart (hunk 3 above)
Comment (HansMuller) [inline packages/flutter/lib/src/material/grid_tile_bar.dart]: You're right, if there's no trailing widget the right pad should be 16, not 8.

**Comment by abarth** [comment]
LGTM

**Pull request merged**